src/neural_operators.jl

```diff
@@ -0,0 +1,101 @@
+abstract type NeuralOperator <: Lux.AbstractExplicitLayer end
+
+"""
+DeepONet(branch,trunk)
+"""
+
+"""
+    DeepONet(branch,trunk,linear=nothing)
+
+`DeepONet` is differential neural operator focused for solving physic-informed parametric ODEs.
+
+DeepONet uses two neural networks, referred to as the "branch" and "trunk", to approximate
+the solution of a differential equation. The branch network takes the spatial variables as
+input and the trunk network takes the temporal variables as input. The final output is
+the dot product of the outputs of the branch and trunk networks.
+
+DeepONet is composed of two separate neural networks referred to as the "branch" and "trunk",
+respectively. The branch net takes on input represents a function  evaluated at a collection
+of fixed locations in some boundsand returns a features embedding. The trunk net takes the
+continuous coordinates as inputs, and outputs a features embedding. The final output of the
+DeepONet, the outputs of the branch and trunk networks are merged together via a dot product.
+
+## Positional Arguments
+*  `branch`: A branch neural network.
+*  `trunk`: A trunk neural network.
+
+## Keyword Arguments
+* `linear`: A linear layer to apply to the output of the branch and trunk networks.
+
+## Example
+
+```julia
+branch = Lux.Chain(
+    Lux.Dense(1, 10, Lux.tanh_fast),
+    Lux.Dense(10, 10, Lux.tanh_fast),
+    Lux.Dense(10, 10))
+trunk = Lux.Chain(
+    Lux.Dense(1, 10, Lux.tanh_fast),
+    Lux.Dense(10, 10, Lux.tanh_fast),
+    Lux.Dense(10, 10, Lux.tanh_fast))
+linear = Lux.Chain(Lux.Dense(10, 1))
+
+deeponet = DeepONet(branch, trunk; linear= linear)
+
+a = rand(1, 50, 40)
+b = rand(1, 1, 40)
+x = (branch = a, trunk = b)
+θ, st = Lux.setup(Random.default_rng(), deeponet)
+y, st = deeponet(x, θ, st)
+```
+
+## References
+* Lu Lu, Pengzhan Jin, George Em Karniadakis "DeepONet: Learning nonlinear operators for identifying differential equations based on the universal approximation theorem of operators"
+* Sifan Wang "Learning the solution operator of parametric partial differential equations with physics-informed DeepOnets"
+"""
+struct DeepONet{L <: Union{Nothing, Lux.AbstractExplicitLayer }} <: NeuralOperator
```

Comment:
This should really be living in NeuralOperators.jl. cc @avik-pal

Reply:
I just implemented it for tests because NeuralOperators.jl was outdate and couldn't used it. Yes agree, it should relocate to NeuralOperators.jl

Reply:
Should it be NeuralOperators.jl or LuxNeuralOperators.jl? I see https://github.com/LuxDL/LuxNeuralOperators.jl/pull/5 implementing Deeponet

Reply:
yes, Lux. I didn't know there was already one like this

Reply:
I can do this task: move DeepOnet from here to LuxNeuralOperators

Reply:
Basic NO and DeepONet are now in LuxNeuralOperators.jl
@KirillZubov